esp/xslt/esdl2monitor.xslt

```diff
@@ -203,32 +204,47 @@ END;
     string100 Message {xpath('Message')};
   END;
 
-  soapoutRec := record
+  soapoutRec := record <xsl:choose><xsl:when test="$platform='ESP'">(the_responseLayout)
+    </xsl:when>
+    <xsl:otherwise>
     dataset (the_responseLayout) ds {xpath('Dataset/Row')};
+    </xsl:otherwise>
+  </xsl:choose>
     exceptionRec Exception {xpath('Exception')};
   end;
 
 MonSoapcall(DATASET(the_requestLayout) req) := FUNCTION
 
-  // Wrap it so that request would look like:
-  // <AssetReportRequest><Row><User>...</User><Options>..</Options><SearchBy>...</SearchBy></Row></AssetReportRequest>
+  <xsl:choose>
+    <xsl:when test="$platform='ESP'">
+  //ESP uses the request layout as is
+  ds_request := req;
+
+    </xsl:when>
+    <xsl:otherwise>
+  // When calling roxie the actual request parameters are placed inside a dataset that is named the same as the request
+  // so it looks like:
+  //   <MyRequest><MyRequest><Row><User>...</User><Options>..</Options><SearchBy>...</SearchBy></Row></MyRequest></MyRequest>
```

Comment:
It's the ECL query standard for handling ESDL based queries.  There is an input dataset that has the same name as the request.  It could have been named differently but we have no control over the standard here.